### PR TITLE
Back to the origin : Cornice

### PR DESCRIPTION
**Pull request by leplatrem**
The branch is on top of #9 

The objective of this PR is to reach the feature set we had with Eve. Also, some patterns might emerge on top of cornice, most probably regarding resource oriented CRUD views

* [x] Cornice boiler plate
* [x] Basic views for device and article resources 
* [x] Memory backend
* [x] Basic authentication
* [x] Basic views for device read status
* [x] FxA authentication
* [x] Record validation
* [x] Hello view
* [x] Heartbeat view
* [x] Fix distinction between 401 and 403 behaviour

**Postponed**
* [ ] Filtering
* [ ] Sorting
* [ ] Pagination (python [paginate](https://github.com/Pylons/paginate) ?)

### CODE REVIEW

**Comment by Natim** [comment]
Closes #3

**Comment by leplatrem** [comment]
@Natim @ametaireau r? :)

Only left 401~403 errors to be clarified...

File: readinglist/authentication.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            raise HTTPServiceUnavailable()
+        except (fxa_errors.InProtocolError, fxa_errors.TrustError):
+            return None
+        return user_id
+
+
+@implementer(IAuthorizationPolicy)
+class AuthorizationPolicy(object):
+    def permits(self, context, principals, permission):
+        return permission in principals
```
Comment (Natim) [inline readinglist/authentication.py]: This looks really strange to me. Permission != principals. The principal is the list of roles not the list of permissions.

**Comment by Natim** [comment]
r+

**Comment by Natim** [comment]
I think we want a travis and tox test matrix :)

**Pull request merged**